ideascube/library/views.py

```diff
@@ -66,6 +66,13 @@ def get_context_data(self, **kwargs):
 
 class BookDetail(DetailView):
     model = Book
+    def get(self, request, *args, **kwargs):
+        ret = super().get(request, *args, **kwargs)
+        user = request.user
+        if user.is_authenticated() and user.is_staff and not self.object.physical:
+            messages.warning(request, _("Please add a specimen, or the book "
+                                        "won't be available for the users"))
```

Comment:
I think a warning is nice: it's already the way we display a message to the user. No need to invent a new one imho.